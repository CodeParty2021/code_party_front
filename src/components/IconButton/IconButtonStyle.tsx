import styled, { css } from "styled-components";
import { iconButton } from "styles/colors";

export type IconButtonStyleProps = {};

const defaultStyle = css`
  display: flex;
  padding: 12px;
  border-radius: 48px;
  background: ${iconButton.bg};
`;

export const IconButtonStyle = styled.button<IconButtonStyleProps>`
  ${defaultStyle}
`;

IconButtonStyle.defaultProps = {};
