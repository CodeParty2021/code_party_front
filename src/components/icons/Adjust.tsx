import React, { ComponentProps } from "react";
import {
  IconPrototype,
  IconsDir,
} from "./components/IconPrototype/IconPrototype";

type Props = {
  size?: 16 | 32;
} & Omit<ComponentProps<typeof IconPrototype>, "filename" | "size">;

const Adjust: React.FC<Props> = ({ size = 16, ...props }) => {
  return (
    <IconPrototype
      filename={`${IconsDir()}/adjust_${size}.svg`}
      size={`${size}px`}
      {...props}
    />
  );
};

export default Adjust;
