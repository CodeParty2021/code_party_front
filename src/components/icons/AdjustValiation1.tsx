import React, { ComponentProps } from "react";
import {
  IconPrototype,
  IconsDir,
} from "./components/IconPrototype/IconPrototype";

type Props = {
  size?: 16 | 32;
} & Omit<ComponentProps<typeof IconPrototype>, "filename" | "size">;

const AdjustValiation1: React.FC<Props> = ({ size = 16, ...props }) => {
  return (
    <IconPrototype
      filename={`${IconsDir()}/adjust_valiation1_${size}.svg`}
      size={`${size}px`}
      {...props}
    />
  );
};

export default AdjustValiation1;
