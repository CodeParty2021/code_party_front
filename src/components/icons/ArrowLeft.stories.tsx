import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import { ArrowLeft } from "./index";

export default {
  title: "Icons/ArrowLeft",
  component: ArrowLeft,
} as ComponentMeta<typeof ArrowLeft>;

const Template: ComponentStory<typeof ArrowLeft> = (args) => (
  <ArrowLeft {...args} />
);

export const Default = Template.bind({});
Default.args = {
  display: "inline-block",
  wrapper: "svg",
  fill: "#252525",
};

export const Orange = Template.bind({});
Orange.args = {
  ...Default.args,
  fill: "#FF5500",
};
