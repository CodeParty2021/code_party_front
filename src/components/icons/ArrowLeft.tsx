import React, { ComponentProps } from "react";
import {
  IconPrototype,
  IconsDir,
} from "./components/IconPrototype/IconPrototype";

type Props = {
  size?: 16 | 24;
} & Omit<ComponentProps<typeof IconPrototype>, "filename" | "size">;

const ArrowLeft: React.FC<Props> = ({ size = 16, ...props }) => {
  return (
    <IconPrototype
      filename={`${IconsDir()}/arrow_left_${size}.svg`}
      size={`${size}px`}
      {...props}
    />
  );
};

export default ArrowLeft;
