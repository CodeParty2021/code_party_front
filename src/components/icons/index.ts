export { default as ArrowLeft } from "./ArrowLeft";
export { default as ArrowRight } from "./ArrowRight";
export { default as Banner } from "./Banner";
export { default as Check } from "./Check";
export { default as CheckCircle } from "./CheckCircle";
export { default as Checked } from "./Checked";
export { default as Link } from "./Link";
export { default as Lock } from "./Lock";
export { default as Plus } from "./Plus";
export { default as Setting } from "./Setting";
export { default as Sword } from "./Sword";
export { default as Times } from "./Times";
export { default as TriangleBottom } from "./TriangleBottom";
export { default as Unlock } from "./Unlock";
