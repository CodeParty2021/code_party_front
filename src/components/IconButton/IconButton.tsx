import React from "react";
import { iconButton } from "styles/colors";
import { Icon24 } from "types/utils";
import { IconButtonStyle, IconButtonStyleProps } from "./IconButtonStyle";

type Props = IconButtonStyleProps & {
  Icon: React.ComponentType<Icon24>;
  onClick?: React.MouseEventHandler<HTMLButtonElement>;
  color?: "blue" | "black";
};

export const IconButton: React.FC<Props> = ({
  Icon,
  onClick,
  color,
  ...styleProps
}) => {
  return (
    <IconButtonStyle onClick={onClick} {...styleProps}>
      {color == "blue" ? (
        <Icon size={24} fill={iconButton.blue} />
      ) : color == "black" ? (
        <Icon size={24} fill={iconButton.black} />
      ) : (
        <Icon size={24} fill={iconButton.blue} /> // 未指定はblue
      )}
    </IconButtonStyle>
  );
};
