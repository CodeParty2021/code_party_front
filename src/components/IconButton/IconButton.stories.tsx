import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";

import { IconButton } from "./IconButton";
import { ArrowLeft } from "components/icons";

export default {
  title: "components/IconButton",
  component: IconButton,
} as ComponentMeta<typeof IconButton>;

const Template: ComponentStory<typeof IconButton> = (args) => (
  <IconButton {...args} />
);

export const Default = Template.bind({});
Default.args = {
  Icon: ArrowLeft,
};
