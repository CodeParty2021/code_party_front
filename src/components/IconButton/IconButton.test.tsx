import React from "react";
import { IconButton } from "./IconButton";
import "jest-styled-components";
import { render, shallow } from "enzyme";
import { ArrowLeft } from "components/icons";

describe("<IconButton />", () => {
  it("snapshot test", () => {
    const wrapper = render(<IconButton Icon={ArrowLeft} />);

    expect(wrapper).toMatchSnapshot();
  });

  it("click test", () => {
    const clickHandler = jest.fn();
    const wrapper = shallow(
      <IconButton Icon={ArrowLeft} onClick={clickHandler} />
    );
    wrapper.simulate("click");
    expect(clickHandler).toHaveBeenCalled();
  });
});
