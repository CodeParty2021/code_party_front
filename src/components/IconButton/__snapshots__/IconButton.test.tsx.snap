// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconButton /> snapshot test 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 12px;
  border-radius: 48px;
  background: #FFF;
}

.c1 {
  display: inline-block;
  width: 24px;
  height: 24px;
  fill: #6074FF;
}

.c1 > span,
.c1 > div,
.c1 > svg {
  display: block;
  width: 100%;
  height: 100%;
}

<button
  class="c0"
>
  <span
    class="c1"
    display="inline-block"
    fill="#6074FF"
  />
</button>
`;
